pkg/templates/machinecontroller/machines.go

```diff
@@ -75,15 +67,14 @@ func createMachineDeployment(cluster *kubeoneapi.KubeOneCluster, workerset kubeo
 		return nil, errors.Wrap(err, "failed to generate machineSpec")
 	}
 
-	config := providerSpec{
```

Comment:
we don't need special `providerSpec` struct, we already have everything in put API, we this change introduce using existing structs.